### ISSUE TITLE
TrilinosFramework:  Sacado needs an updated GTest

### ISSUE THREAD

**Issue by etphipp** [trilinos/Trilinos#7698]
@trilinos/framework 
@trilinos/sacado 

I have been converting a bunch of unit tests in Sacado that previously used CppUnit to GTest.  The main driver for this is you had to have the CppUnit TPL installed which almost no one has, so these tests were rarely run, and certainly were never run as part of the PR tester nor nightly testing.  However the version of GTest included in Trilinos/commonTools/gtest is not sufficient.  In particular, Sacado needs the "type-parameterized test" capability since almost all of these tests are templated on the Sacado AD type.  However the version in Trilinos appears to not support this capability.  It is unclear to me if this is because that capability didn't exist in GTest at the time it was added, or if only a subset of the GTest functionality was included in Trilinos (in the stock GTest on github, this functionality appears in a separate header that Trilinos does not have).

Can the version of GTest in Trilinos be updated to include this functionality?  Or should I just commit my own version of GTest in Sacado?

**Comment by etphipp** [comment]
No response, so I guess I'll just add my own version of GTest in Sacado.

**Comment by prwolfe** [comment]
Please do not duplicate this.

The version in use is under commonTools and I expect if you replace that and run everyones tests the results will get approved. Not sure how this got missed.

**Comment by etphipp** [comment]
OK, I can try moving the version I added in Sacado to commonTools, replacing what is there.  My only concern is this version is not header-only like the one currently there.  Will that cause problems for downstream customers (e.g., Sierra)?  FYI, the relevant PR for this is #7736.

**Comment by etphipp** [comment]
I was unable to get Trilinos' installed CMake files to work correctly with the new version of GTest I added in commonTools.  Basically the issue was Trilinos' CMake targets file wasn't pulling in the targets file from the nested googletest library.  The only way I could see to fix it would be to write a custom CMake targets file for the Trilinos GTest package, which I don't know how to do.  So I ended up just putting the version of GTest I need in Sacado as a test-only library.  If someone else wants to pick this up and can solve the CMake targets issue, I would be happy remove the version in Sacado.
